src/java/voldemort/utils/StoreDefinitionUtils.java

```diff
@@ -281,41 +281,62 @@ public static void validateNewStoreDefsAreNonBreaking(List<StoreDefinition> oldS
      *
      * Non-breaking changes include changes to
      *   description
-     *   replicationFactor
      *   preferredWrites
      *   requiredWrites
      *   preferredReads
      *   requiredReads
      *   retentionPeriodDays
      *   retentionScanThrottleRate
-     *   retentionFrequencyDays -
+     *   retentionFrequencyDays
+     *   viewOf
      *   zoneCountReads
      *   zoneCountWrites
      *   owners
      *   memoryFootprintMB
      *
+     * non breaking changes also include changes to the serializer type (name field) for
+     *   keySerializer
+     *   valueSerializer
+     *   transformSerializer
+     *
      * @param oldStoreDef
      * @param newStoreDef
      */
     public static void validateNewStoreDefIsNonBreaking(StoreDefinition oldStoreDef, StoreDefinition newStoreDef){
         if (!oldStoreDef.getName().equals(newStoreDef.getName())){
             throw new VoldemortException("Cannot compare stores of different names: " + oldStoreDef.getName() + " and " + newStoreDef.getName());
         }
-      String store = oldStoreDef.getName();
+        String store = oldStoreDef.getName();
+        verifySamePropertyForUpdate(oldStoreDef.getReplicationFactor(), newStoreDef.getReplicationFactor(), "ReplicationFactor", store);
         verifySamePropertyForUpdate(oldStoreDef.getType(), newStoreDef.getType(), "Type", store);
-        verifySamePropertyForUpdate(oldStoreDef.getKeySerializer(), newStoreDef.getKeySerializer(), "KeySerializer", store);
```

Comment:
Can we conditionally execute the key/value serializer checks only for Read-Only stores?